substrate/frame/support/src/lib.rs

```diff
@@ -1019,1221 +916,840 @@ pub mod pallet_prelude {
+	pub use frame_support_procedural::getter;
+
+	/// DEPRECATED: Will be removed, do not use.
+	/// See <https://github.com/paritytech/substrate/pull/13535> for more details.
+	///
+	/// To generate a `Store` trait associating all storages, annotate your `Pallet` struct
+	/// with the attribute `#[pallet::generate_store($vis trait Store)]`, e.g.:
+	///
+	/// ```ignore
+	/// #[pallet::pallet]
+	/// #[pallet::generate_store(pub(super) trait Store)]
+	/// pub struct Pallet<T>(_);
+	/// ```
+	/// More precisely, the `Store` trait contains an associated type for each storage. It is
+	/// implemented for `Pallet` allowing access to the storage from pallet struct.
+	///
+	/// Thus when defining a storage named `Foo`, it can later be accessed from `Pallet` using
+	/// `<Pallet as Store>::Foo`.
+	///
+	/// NOTE: this attribute is only valid when applied _directly_ to your `Pallet` struct
+	/// definition.
+	pub use frame_support_procedural::generate_store;
+
+	/// Allows you to define some extra constants to be added into constant metadata.
+	///
+	/// Must be defined like:
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	/// 	use frame_support::pallet_prelude::*;
+	///
+	/// 	#[pallet::pallet]
+	/// 	pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::config]
+	/// 	#[pallet::disable_frame_system_supertrait_check]
+	/// 	pub trait Config: frame_system::Config {}
+	///
+	/// 	#[pallet::extra_constants]
+	/// 	impl<T: Config> Pallet<T> // $optional_where_clause
+	/// 	{
+	/// 	#[pallet::constant_name(SomeU32ConstantName)]
+	/// 		/// Some doc
+	/// 		fn some_u32_constant() -> u32 {
+	/// 			100u32
+	/// 		}
+	/// 	}
+	/// }
+	/// ```
+	///
+	/// I.e. a regular rust `impl` block with some optional where clause and functions with 0
+	/// args, 0 generics, and some return type.
+	pub use frame_support_procedural::extra_constants;
+
+	#[rustfmt::skip]
+	/// To bypass the `frame_system::Config` supertrait check, use the attribute
+	/// `pallet::disable_frame_system_supertrait_check`, e.g.:
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	/// 	use frame_support::pallet_prelude::*;
+	///
+	/// 	#[pallet::pallet]
+	/// 	pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::config]
+	/// 	#[pallet::disable_frame_system_supertrait_check]
+	/// 	pub trait Config: frame_system::Config {}
+	/// }
+	/// ```
+	///
+	/// Bypassing the `frame_system::Config` supertrait check is typically desirable when
+	/// you want to write an alternative to the `frame_system` pallet.
+	///
+	/// To learn more about supertraits, see the
+	/// [trait_based_programming](../../polkadot_sdk_docs/reference_docs/trait_based_programming/index.html)
+	/// reference doc.
+	pub use frame_support_procedural::disable_frame_system_supertrait_check;
+
+	/// The mandatory attribute `#[pallet::config]` defines the configurable options for the
+	/// pallet.
+	///
+	/// Item must be defined as:
+	///
+	/// ```
+	/// #[frame_support::pallet]
```

Comment:
fwiw, you generally can try and hide imports or useless lines by prepending them with `#`.

Reply:
Didn't know this was possible, super helpful thanks.